Source/WebCore/dom/Document.idl

```diff
@@ -41,6 +41,8 @@ typedef (HTMLScriptElement or SVGScriptElement) HTMLOrSVGScriptElement;
 ] interface Document : Node {
     [CallWith=CurrentDocument] constructor();
 
+    [CallWith=CurrentScriptExecutionContext, EnabledBySetting=DeclarativeShadowRootsParserAPIsEnabled] static Document parseHTMLUnsafe(DOMString html);
```

Comment:
After initial review I'll clean this up to use `CurrentDocument` which also avoids a downcast in `Document`.
```suggestion
    [CallWith= CurrentDocument, EnabledBySetting=DeclarativeShadowRootsParserAPIsEnabled] static Document parseHTMLUnsafe(DOMString html);
```